### PR TITLE
Better "always true/false" errors

### LINKED ISSUE #25245: Confusing error message: Operator '===' cannot be applied to types '1' and '0'.
**TypeScript Version:**  3.0.0-dev.20180623

**Search Terms:** 

**Code**

```ts
function foo(x: [number]) {
    return x.length === 0;
}
```

**Bug:**
The error message is: `Operator '===' cannot be applied to types '1' and '0'.`

I found this error very confusing. It took me a while to understand that the error meant that the type on the left hand side was `1`.

In this case, I think we should instead report that the condition always evaluates to false.

**Related Issues:** 
- #11178

/cc @chrisdias

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #25245

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@mhegazy @RyanCavanaugh any idea why the tsserver test fails?

**Comment by s-ve** [comment]
@DanielRosenwasser 
```
The types of these values indicates that this condition will always be...
```

Shoudn't it be "**indicate**" ? (without the "s")

**Comment by DanielRosenwasser** [comment]
Thanks for indicating that typo.

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -1252,6 +1252,10 @@
         "category": "Error",
         "code": 2366
     },
+    "The types of these values indicate that this condition will always be '{0}'.": {
```
Comment (mihailik) [inline src/compiler/diagnosticMessages.json]: If those are statically known primitive types, can we include actual values in error message? Like:

`"These values are known to be '{0}' and '{1}', rendering this condition always '{2}'."`

File: tests/baselines/reference/capturedLetConstInLoop5.errors.txt [hunk 2]
```diff
@@ -235,7 +238,8 @@ tests/cases/compiler/capturedLetConstInLoop5.ts(229,13): error TS2365: Operator
             (() => x + y + v);
             if (x == 1) {
                 ~~~~~~
-!!! error TS2365: Operator '==' cannot be applied to types '0' and '1'.
+!!! error TS2367: The types of these values indicate that this condition will always be 'false'.
```
Comment (mihailik) [inline tests/baselines/reference/capturedLetConstInLoop5.errors.txt]: `These values are known to be '0' and '1', rendering this condition always 'false'.`
 ^^^^ how much smoother to comprehend and fix!

File: src/compiler/diagnosticMessages.json (hunk 1 above)
Comment (DanielRosenwasser) [reply src/compiler/diagnosticMessages.json]: Check out #25311

**Pull request merged**